build.gradle

```diff
@@ -222,14 +186,62 @@ task runTestsWithTag(type: Test) {
             mkdir "${project.buildDir}/test-working-dir/"
             systemProperty "TERRA_WORKING_DIR", "${project.buildDir}/test-working-dir/"
         }
+        // Integration tests wrap various external command-line tools which often maintain their own global state.
+        // Running them in parallel will cause runners to clobber this state, so these tests must be run serially.
+        maxParallelForks = (testTag == "unit" ? 12 : 1)
+    }
+    beforeTest { descriptor ->
+        println "========================================================="
+        println "Running test: ${descriptor.name} [${descriptor.className}]"
+    }
+    afterTest { descriptor, result ->
+        println "Finished running test ${descriptor.name} [${descriptor.className}] with result: ${result.resultType}"
+    }
+    testLogging {
+        showStandardStreams = true // disable stdout & stderr redirection
+    }
 
-        // generate a unique id for each test run
-        systemProperty "TEST_RUN_ID", UUID.randomUUID().toString()
+    boolean contextDir = project.hasProperty("contextDir")
+
+    // by default, tests override the global context directory with a sub-directory of the gradle build directory
+    // this makes it easier to clean up after tests and makes sure the tests don't overwrite any context that
+    // exists on this same machine. you can override this default behavior by specifying -PcontextDir=/path/to/dir.
+    // this will cause context to be written to the specified directory.
+    if (!contextDir) {
+        environment "TERRA_CONTEXT_PARENT_DIR", "${buildDir}/test-context/"
```

Comment:
optional random thought I had: To make things simpler, stop supporting CONTEXT_DIR_OVERRIDE_NAME